src/gui/Lv2ViewBase.cpp

```diff
@@ -51,13 +52,51 @@ namespace lmms::gui
 {
 
 
-Lv2ViewProc::Lv2ViewProc(QWidget* parent, Lv2Proc* ctrlBase, int colNum) :
-	LinkedModelGroupView (parent, ctrlBase, colNum)
+static int numDigitsAsInt(float f)
 {
-	class SetupWidget : public Lv2Ports::ConstVisitor
+	// use rounding:
+	// LcdSpinBox sometimes uses roundf(), sometimes cast rounding
+	// we use rounding to be on the "safe side"
+	const float rounded = roundf(f);
+	int asInt = static_cast<int>(rounded);
+	int digits = 1; // always at least 1
+	if(asInt < 0)
+	{
+		++digits;
+		asInt = -asInt;
+	}
+	// "asInt" is positive from now
+	int power = 10;
+	for(int i = 1; i<6; ++i, power *= 10)
+	{
+		if(asInt >= power) { ++digits; } // 2 digits for >=10, 3 for >=
+		else { break; }
+	}
+	return digits;
+}
+
+
+
+
+Lv2ViewProc::Lv2ViewProc(QWidget* parent, Lv2Proc* proc, int colNum) :
+	LinkedModelGroupView (parent, proc, colNum)
+{
+//#define TEST_NUM_DIGITS_AS_INT
+#ifdef TEST_NUM_DIGITS_AS_INT
```

Comment:
Is this a really good place to test the function?

Reply:
What about `LcdSpinBox.cpp`? Or do you see other applications outside of LcdSpinBox, which justifies putting it into the core?